operators/src/engine/execution_context.rs

```diff
@@ -0,0 +1,34 @@
+use crate::concurrency::{ThreadPool, ThreadPoolContext};
+use std::path::PathBuf;
+
+/// A context that provides certain utility access during operator execution
+/// TODO: maybe this is rather a trait
+#[derive(Debug, Clone)]
+pub struct ExecutionContext<'pool> {
+    pub raster_data_root: PathBuf,
+    pub thread_pool: ThreadPoolContext<'pool>,
+}
+
+/// Create a provider for execution contexts in test environments
+pub struct MockExecutionContextCreator {
+    raster_data_root: PathBuf,
+    thread_pool: ThreadPool,
+}
+
+impl MockExecutionContextCreator {
```

Comment:
what is the purpose of this creator? Where do you use it more than once to create multiple contexts? For the QueryContext I defined a trait and if you would do it like this here the MockExecutionContext could just own the threadpool. Would also be nice if the different Contexts would be implemented similarily

Reply:
I just need something to give me a `ThreadPoolContext` with its lifetime. I guess I can't put the `ThreadPool` and `ThreadPoolContext` into the same struct since `ThreadPoolContext` contains a reference to `ThreadPool`.

The trait idea sounds okay, but I don't know if it is part of this PR. There are many questions regarding the `ExecutionContext`, e.g. raster paths and configurations.

Reply:
ok let's do a separate PR for this